clang/test/SemaSYCL/fpga_pipes_ast.cpp

```diff
@@ -0,0 +1,56 @@
+// RUN: %clang_cc1 -fsycl-is-device -fsyntax-only -ast-dump %s | FileCheck %s
+
+// Tests for AST of __attribute__((io_pipe_id(..))) attribute.
+
+struct pipe_storage {};
+
+// CHECK: VarDecl {{.*}} Storage1 'const pipe_storage' callinit
+// CHECK-NEXT: CXXConstructExpr {{.*}} 'const pipe_storage' 'void () noexcept'
+// CHECK-NEXT: SYCLIntelPipeIOAttr
+// CHECK-NEXT: ConstantExpr{{.*}}'int'
+// CHECK-NEXT: value: Int 5
+// CHECK-NEXT: IntegerLiteral {{.*}} 'int' 5
+const pipe_storage Storage1 __attribute__((io_pipe_id(5)));
+
+// CHECK: VarTemplateDecl {{.*}} Storage2
+// CHECK: VarDecl {{.*}} Storage2 'pipe_storage' callinit
+// CHECK-NEXT: CXXConstructExpr {{.*}} 'pipe_storage' 'void () noexcept'
+// CHECK_NEXT: SYCLIntelPipeIOAttr
+// CHECK_NEXT: DeclRefExpr {{.*}} 'int' NonTypeTemplateParm {{.*}} 'N' 'int'
+// CHECK: VarTemplateSpecializationDecl {{.*}} used Storage2 'pipe_storage' callinit
+// CHECK-NEXT: TemplateArgument integral 2
+// CHECK-NEXT: CXXConstructExpr {{.*}} 'pipe_storage' 'void () noexcept'
+// CHECK-NEXT: SYCLIntelPipeIOAttr
+// CHECK-NEXT: ConstantExpr{{.*}}'int'
+// CHECK-NEXT: value: Int 2
+// CHECK-NEXT: SubstNonTypeTemplateParmExpr {{.*}} 'int'
+// CHECK-NEXT: NonTypeTemplateParmDecl {{.*}} referenced 'int' depth 0 index 0 N
+// CHECK-NEXT: IntegerLiteral {{.*}} 'int' 2
+template <int N>
+pipe_storage Storage2 __attribute__((io_pipe_id(N)));
+
+int main() {
+  Storage2<2>;
+  return 0;
+}
+
+// Checking of different argument values.
+// CHECK: VarDecl {{.*}} Storage3 'const pipe_storage' callinit
+// CHECK-NEXT: CXXConstructExpr {{.*}} 'const pipe_storage' 'void () noexcept'
+// CHECK-NEXT: SYCLIntelPipeIOAttr
+// CHECK-NEXT: ConstantExpr{{.*}}'int'
+// CHECK-NEXT: value: Int 4
+// CHECK-NEXT: IntegerLiteral {{.*}} 'int' 4
+const pipe_storage Storage3 __attribute__((io_pipe_id(4))) // expected-note{{previous attribute is here}}
```

Comment:
Since you do not have `-verify ` in test line, this is not actually testing diagnostics. I don't think you need a diagnostic test here though since we already do that in another test,

If you want to keep this test, you can remove the `expected-*` and added a CHECK-NOT to ensure the conflicting attribute is not added to the AST. 

Same point for test below - maybe you can add a CHECK-NOT

Reply:
>>Since you do not have -verify in test line, this is not actually testing diagnostics. I don't think you need a diagnostic test here though since we already do that in another test,

Thanks for catching this. I have removed this test case since we already do that in another test.

>Same point for test below - maybe you can add a CHECK-NOT

I have added CHECK-NOT for this test.